drumkv1: 0.9.23 -> 1.0.0 (+fix LV2 Qt UI issues)

TheRedstoneDEV-DE opened the pull request:
## Description of changes
Updated the drumkv1 drum sampler from version 0.9.23 to 1.0.0 (changelog: https://drumkv1.sourceforge.io/drumkv1-downloads.html, couldn't find any better one) and added xorg.libX11 to the dependencies. 
As this is my first pull-request I hope I didn't do anything completely wrong, met all guidelines and don't cause a huge waste of time.
<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done
- changed version to 1.0.0 and added corresponding checksum
- added xorg.libX11to dependencies (it is used to enable LV2 X11 UI support)
- updated nix file in general (was not able to be built before)
- added wrapQtAppsHook
- update to Qt6 dependency
- added a cmake flag to disable the  experimental feature "LV2 port change request"
- added workaround to get it to package correctly
- (Wayland support remains disabled, because it is experimental)
#### After I found out that the original maintainer was gone:
- add myself to maintainer-list.nix
- add myself as maintainer of this package
<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

TheRedstoneDEV-DE committed:
drumkv1: 0.9.23 -> 1.0.0 (+fix LV2 Qt UI issues)

TheRedstoneDEV-DE commented:
added myself as maintainer of the package since the original maintainer ("goibhniu") is gone

SuperSandro2000 commented on pkgs/applications/audio/drumkv1/default.nix:
```diff
@@ -21,4 +56,4 @@ mkDerivation rec {
     maintainers = [ maintainers.goibhniu ];
   };
-}
+}
```
Please add the final newline back and configure your editor to always add that.

SuperSandro2000 commented on pkgs/applications/audio/drumkv1/default.nix:
```diff
@@ -1,17 +1,52 @@
+  cmakeFlags = [ "-DCONFIG_LV2_PORT_CHANGE_REQUEST=false" ]; # disable experimental feature "LV2 port change request"
+
+  installPhase = let
+  lv2Dir = "$out/lib/lv2/drumkv1.lv2";
```
```suggestion
    lv2Dir = "$out/lib/lv2/drumkv1.lv2";
```

SuperSandro2000 commented on pkgs/applications/audio/drumkv1/default.nix:
```diff
@@ -1,17 +1,52 @@
+    runHook preInstall
+    make install
+    mkdir -p ${lv2Dir}
+    # -- some workaround, since it always wants to install to /$out/$out -- #
```
That means the install location in the cmake file is wrong. Please fix that. It likely needs FULL_INSTALL_DIR or so instead. This should also be upstreamed

SuperSandro2000 commented on pkgs/applications/audio/drumkv1/default.nix:
```diff
@@ -1,17 +1,52 @@
+    mkdir -p ${lv2Dir}
+    # -- some workaround, since it always wants to install to /$out/$out -- #
+    mv $out/${lv2Dir} $out/lib/lv2/
+    runHook postInstall
+    rm -r $out/nix
```
```suggestion

    rm -r $out/nix
    runHook postInstall
```

SuperSandro2000 commented on pkgs/applications/audio/drumkv1/default.nix:
```diff
@@ -1,17 +1,52 @@
+  installPhase = let
+  lv2Dir = "$out/lib/lv2/drumkv1.lv2";
+  in ''
+    runHook preInstall
+    make install
```
We should move everything into postInstall and avoid calling make install by hand.

SuperSandro2000 commented on pkgs/applications/audio/drumkv1/default.nix:
```diff
@@ -1,17 +1,52 @@
-  nativeBuildInputs = [ pkg-config ];
+  nativeBuildInputs = [ pkg-config cmake qt6.wrapQtAppsHook ];
+
+  cmakeFlags = [ "-DCONFIG_LV2_PORT_CHANGE_REQUEST=false" ]; # disable experimental feature "LV2 port change request"
```
```suggestion
  cmakeFlags = [
    # disable experimental feature "LV2 port change request"
    "-DCONFIG_LV2_PORT_CHANGE_REQUEST=false"
  ];
```

SuperSandro2000 commented on pkgs/applications/audio/drumkv1/default.nix:
```diff
@@ -1,17 +1,52 @@
+    libsndfile 
+    liblo 
+    lv2 
+    qt6.full 
```
Please list the package individually instead of relying on the big qt6.full

SuperSandro2000 commented on pkgs/applications/audio/drumkv1/default.nix:
```diff
@@ -1,17 +1,52 @@
+  version = "1.0.0";
 
   src = fetchurl {
     url = "mirror://sourceforge/drumkv1/${pname}-${version}.tar.gz";
```
```suggestion
    url = "mirror://sourceforge/drumkv1/drumkv1-${version}.tar.gz";
```

SuperSandro2000 commented on pkgs/applications/audio/drumkv1/default.nix:
```diff
@@ -1,4 +1,4 @@
-{ 
```
This should be a squashed in the previous commit

SuperSandro2000 commented on pkgs/applications/audio/drumkv1/default.nix:
This can also be done in the initial commit

SuperSandro2000 reviewed (changes requested):
Please drop the merge commit from this PR.

TheRedstoneDEV-DE committed:
maintainers: add theredstonedev

TheRedstoneDEV-DE commented:
I now reworked my commits (and dropped the merge), I hope I implemented everything correctly...
I have also done some more research on the problem I had and created an issue at the upstream. I also implemented a more elegant, temporary fix that can be used until upstream fixed the problem (It had to do with "CMAKE_INSTALL_LIBDIR" being set to a absolute path, which was not accounted for in upstream's CMakeLists).

SuperSandro2000 commented:
Please rebase on master

TheRedstoneDEV-DE commented:
Ah, thanks for reminding me, I didn't have collisions in a PR before...
I hope everything is fine now :)

FliegendeWurst commented on maintainers/maintainer-list.nix:
```diff
@@ -22207,6 +22207,13 @@
     name = "Rouven Seifert";
     keys = [ { fingerprint = "1169 87A8 DD3F 78FF 8601  BF4D B95E 8FE6 B11C 4D09"; } ];
   };
+  theredstonedev = {
```
This commit should be named `maintainers: add theredstonedev`.
LGTM otherwise

SuperSandro2000 commented on pkgs/applications/audio/drumkv1/default.nix:
```diff
@@ -1,24 +1,58 @@
-    url = "mirror://sourceforge/drumkv1/${pname}-${version}.tar.gz";
-    sha256 = "sha256-gNscsqGpEfU1CNJDlBAzum9M0vzJSm6Wx5b/zhOt+sk=";
+    url = "mirror://sourceforge/drumkv1/drumkv1-${version}.tar.gz";
+    sha256 = "sha256-vi//84boqaVxC/KCg+HF76vB4Opch02LU4RtbVaxaX4=";
```
```suggestion
    hash = "sha256-vi//84boqaVxC/KCg+HF76vB4Opch02LU4RtbVaxaX4=";
```

SuperSandro2000 reviewed (commented):
Otherwise LGTM

TheRedstoneDEV-DE commented:
Everything should be fixed now...
